test/com/google/enterprise/adaptor/ValidatedUriTest.java

```diff
@@ -0,0 +1,157 @@
+
+import static org.junit.Assert.assertEquals;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.logging.Handler;
+import java.util.logging.Level;
+import java.util.logging.LogRecord;
+import java.util.logging.Logger;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+/** Unit tests for {@link ValidatedUri}. */
+public class ValidatedUriTest {
+
+  @Rule
+  public ExpectedException thrown = ExpectedException.none();
+
+  @Test
+  public void testNullUrl() throws Exception {
+    thrown.expect(URISyntaxException.class);
+    new ValidatedUri(null);
+  }
+
+  @Test
+  public void testEmptyUrl() throws Exception {
+    thrown.expect(URISyntaxException.class);
+    new ValidatedUri("");
+  }
+
+  @Test
+  public void testNoProtocol() throws Exception {
+    thrown.expect(URISyntaxException.class);
+    new ValidatedUri("//foo/bar");
+  }
+
+  @Test
+  public void testUnknownProtocol() throws Exception {
+    thrown.expect(URISyntaxException.class);
+    new ValidatedUri("unknown://foo/bar");
+  }
+
+  @Test
+  public void testBadProtocol() throws Exception {
+    thrown.expect(URISyntaxException.class);
+    new ValidatedUri("https//foo/bar");
+  }
+
+  @Test
+  public void testNoHost() throws Exception {
+    thrown.expect(URISyntaxException.class);
+    new ValidatedUri("http://");
+  }
+
+  @Test
+  public void testRelativeUri() throws Exception {
+    thrown.expect(URISyntaxException.class);
+    new ValidatedUri("foo/bar");
+  }
+
+  @Test
+  public void testMessageFormatRemnants() throws Exception {
+    thrown.expect(URISyntaxException.class);
+    new ValidatedUri("http://message_format/foo/{0}");
+  }
+
+  @Test
+  public void testMessageFormatRemnantsInQuery() throws Exception {
+    thrown.expect(URISyntaxException.class);
+    new ValidatedUri("http://message_format/foo?q={0}");
+  }
+
+  @Test
+  public void testNakedIPv6Address() throws Exception {
+    thrown.expect(URISyntaxException.class);
+    new ValidatedUri("http://::1/foo/bar");
+  }
+
+  @Test
+  public void testBracketedIPv6Address() throws Exception {
+    assertEquals(new URI("http://[::1]/foo/bar"),
+        new ValidatedUri("http://[::1]/foo/bar").getUri());
+  }
+
+  @Test
+  public void testRootPath() throws Exception {
+    assertEquals(new URI("http://foo:80/"),
+        new ValidatedUri("http://foo:80/").getUri());
+  }
+
+  @Test
+  public void testGetUri() throws Exception {
+    assertEquals(new URI("http://example.com/foo/bar"),
+        new ValidatedUri("http://example.com/foo/bar").getUri());
+  }
+
+  @Test
+  public void testReachableHost() throws Exception {
+    List<String> messages = new ArrayList<String>();
+    captureLogMessages(ValidatedUri.class, "is not reachable", messages);
+    new ValidatedUri("http://127.0.0.1/foo/bar").logIfHostIsNotReachable();
+    assertEquals(0, messages.size());
```

Comment:
Maybe assertEquals(Collections.<String>emptyList(), messages) or at least (for consistency with the below), assertEquals(messages.toString(), 0, messages.size())?